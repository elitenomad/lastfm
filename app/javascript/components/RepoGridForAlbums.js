import React from 'react';
import PropTypes from "prop-types";

function RepoGridForAlbums ({ albums }) {
    return (
        <ul className='popular-list'>
            {albums.map(({name, image,playcount, url, artist}, index) => {
                return (
                    <li key={name} className='popular-item'>
                        <div className='popular-rank'>#{index + 1}</div>
                        <ul className='space-list-items'>
                            <li>
                                <img
                                    className='avatar'
                                    src={image[2]['#text'] || '/img/No_album_cover.jpg'}
                                    alt={`Avatar for ${artist.name}`}
                                />
                            </li>
                            <li>{name}</li>
                            <li>{ url }</li>
                            <li>@{artist.name}</li>
                            <li>@{artist.url}</li>
                        </ul>
                    </li>
                )
            })}
        </ul>
    )
}

RepoGridForAlbums.propTypes = {
    albums: PropTypes.array.isRequired,
};


export default RepoGridForAlbums;