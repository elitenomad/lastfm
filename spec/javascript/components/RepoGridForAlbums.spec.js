import RepoGridForAlbums from './../../../app/javascript/components/RepoGridForAlbums';
import React from 'react';
import { shallow } from 'enzyme';

describe('RepoGridForAlbums Component', () => {
    const albums = [
        {
            "name": "Believe",
            "playcount": 2007078,
            "mbid": "63b3a8ca-26f2-4e2b-b867-647a6ec2bebd",
            "url": "https://www.last.fm/music/Cher/Believe",
            "artist": {
                "name": "Cher",
                "mbid": "bfcc6d75-a6a5-4bc6-8282-47aec8531818",
                "url": "https://www.last.fm/music/Cher"
            },
            "image": [
                {
                    "#text": "https://lastfm-img2.akamaized.net/i/u/34s/b0c2311a9af7f0edbc8b99450944ca1b.png",
                    "size": "small"
                },
                {
                    "#text": "https://lastfm-img2.akamaized.net/i/u/64s/b0c2311a9af7f0edbc8b99450944ca1b.png",
                    "size": "medium"
                },
                {
                    "#text": "https://lastfm-img2.akamaized.net/i/u/174s/b0c2311a9af7f0edbc8b99450944ca1b.png",
                    "size": "large"
                },
                {
                    "#text": "https://lastfm-img2.akamaized.net/i/u/300x300/b0c2311a9af7f0edbc8b99450944ca1b.png",
                    "size": "extralarge"
                }
            ]
        }
    ];

    it('is expected to render correctly', () => {
        const wrapper = shallow(<RepoGridForAlbums  albums={albums}/>);
        expect(wrapper).toMatchSnapshot();
        const li = wrapper.find('.poplular-item');
        expect(li.find('ul.space-list-items')).not.toBeNull();
    });
});


