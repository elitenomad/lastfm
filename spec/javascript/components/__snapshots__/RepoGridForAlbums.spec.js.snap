// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoGridForAlbums Component is expected to render correctly 1`] = `
<ul
  className="popular-list"
>
  <li
    className="popular-item"
    key="Believe"
  >
    <div
      className="popular-rank"
    >
      #
      1
    </div>
    <ul
      className="space-list-items"
    >
      <li>
        <img
          alt="Avatar for Cher"
          className="avatar"
          src="https://lastfm-img2.akamaized.net/i/u/174s/b0c2311a9af7f0edbc8b99450944ca1b.png"
        />
      </li>
      <li>
        Believe
      </li>
      <li>
        https://www.last.fm/music/Cher/Believe
      </li>
      <li>
        @
        Cher
      </li>
      <li>
        @
        https://www.last.fm/music/Cher
      </li>
    </ul>
  </li>
</ul>
`;
